libvast/vast/system/node_state.hpp

```diff
@@ -0,0 +1,24 @@
+#ifndef VAST_SYSTEM_NODE_STATE_HPP
+#define VAST_SYSTEM_NODE_STATE_HPP
+
+#include "vast/expression.hpp"
+#include "vast/filesystem.hpp"
+
+#include "vast/system/tracker.hpp"
+
+namespace vast {
+namespace system {
+
+/// A container for VAST components.
+struct node_state {
+  path dir;
+  tracker_type tracker;
+  std::unordered_map<std::string, int> labels;
+  std::string name = "node";
+  std::unordered_map<expression, caf::actor> continuous_queries;
```

Comment:
> Don't think this is currently in use. Wanted to add booking to allow the NODE to pass existing queries to IMPORTERs that are spawned later on.

Yeah, that's conceptually the right idea. But TRACKER should really do this form of bookkeeping. When the IMPORTER-after-EXPORTER scenario occurs, NODE should simply ask the TRACKER to provide the necessary information.

To do so, we'd have to patch TRACKER so that it keeps track of the EXPORTER type. Let's keep that in mind for the future.